### PR TITLE
docs: HTML API changes for charts/forms PTL-1521

### PR DESCRIPTION
**Pull request by SzymonZur**
ATTENTION! NEW DOCS STRUCTURE
FOR THE NEXT THREE WEEKS, ONLY BRANCH FROM new-docs-structure
OTHERWISE, WAIT UNTIL THE NEW STRUCTURE HAS BEEN RELEASED

  - Please check the [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4) for details
  
__________

Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?

Have you done a trial build to check all new or changed links?

Is there anything else you would like us to know?

**This week's exciting advice from the style guide**

- Write your headings in sentence case:

  - A good example
    This is a correct heading.
  - A Bad Example
    This is not a correct heading.

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/01_chart-types.mdx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 title: 'G2Plot Chart Types'
-sidebar_label: 'Chart types'
+sidebar_label: 'Chart Types'
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/01_chart-types.mdx:3]: As much as I personally like this case, the direction from learn is that we use setence case for titles of pages and sections. Same in all other places
```suggestion
sidebar_label: 'Chart types'
```

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+title: 'Chart Declerative HTML API'
+sidebar_label: 'Chart Declerative HTML'
+id: chart-html-api
+keywords: [genesis, foundation, ui, chart, g2plot-chart, g2plot]
+tags:
+  - connected
+  - datasource
+  - chart
+  - g2plot-chart
+  - g2plot
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx:11]: Can we expand these sections please with anything you can think of relating to the API. These terms aid searchabiliy, which is something we've been critisised for previously

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+
+</td>
+    </tr>
+  </tbody>
+</table>
+
+#### Slots
+
+| Slot | Use |
+|---|---|
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx:97]: For completeness sake we should add the Default slot here too, and say that is where the chart datasource goes (and you can do an internal page link to that section)

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+  </thead>
+  <tbody>
+    <tr>
+      <td>request</td>
+      <td><code>any</code></td>
+      <td>This optional parameter enables you to specify request fields, which can include wildcards..</td>
+<td>
+
+```typescript
+<chart-datasource :request=${(x) => x.datasourceRequest}>
```
Comment (patrickoneill-genesis) [inline docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx:394]: Given this is an any, can we have a concrete example of what `datasourceRequest` would be?

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+```typescript
+<chart-datasource resourceName="ALL_TRADES">
+```
+
+</td>
+        </tr>
+        <tr>
+            <td>criteria</td>
+            <td><code>string</code></td>
+            <td>Clients can send a Groovy expression to perform filters on the query server, these remain active for the life of the subscription.</td>
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx:231]: Is there a link to any server pages which cover writing these filters in more detail we could link to here?

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+      <th>Type</th>
+      <th>Use</th>
+      <th>Example</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>request</td>
+      <td><code>any</code></td>
+      <td>This optional parameter enables you to specify request fields, which can include wildcards..</td>
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx:390]:
```suggestion
      <td>This optional parameter enables you to specify request fields, which can include wildcards.</td>
```

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx (hunk 4 above)
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx:394]: Can we add an additional part to this example showing an example x.datasourceRequest, with the type being `any` I don't think it's 100% clear how to use this

**Review by matteematt — COMMENTED** [review]
Great work, just a few small points from me

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/01_chart-types.mdx (hunk 1 above)
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/01_chart-types.mdx]: [chore: addressed PR feedback v1 PTL-1521](https://github.com/genesiscommunitysuccess/docs/pull/1918/commits/7cc3d59862e0ce6d0c1fdb0790b0374f34dbdd8f)

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx (hunk 5 above)
Comment (matteematt) [reply docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx:231]: @SzymonZur I have reached out to Patrick about this on Slack and he's going to create a reuseable section on this we can include in appropriate pages

**Review by matteematt — APPROVED** [review]

**Pull request merged**